keras_cv/layers/preprocessing3d/frustum_random_point_feature_noise.py

```diff
@@ -0,0 +1,121 @@
+    The point_clouds tensor shape must be specific and cannot be dynamic.
+    During inference time, the output will be identical to input. Call the layer with `training=True` to add noise to the input points.
+
+    Input shape:
+      point_clouds: 3D (multi frames) float32 Tensor with shape
+        [num of frames, num of points, num of point features].
+        The first 5 features are [x, y, z, class, range].
+      bounding_boxes: 3D (multi frames) float32 Tensor with shape
+        [num of frames, num of boxes, num of box features].
+        The first 7 features are [x, y, z, dx, dy, dz, phi].
+
+    Output shape:
+      A dictionary of Tensors with the same shape as input Tensors.
+
+    Arguments:
+      r_distance: A float scalar sets the starting distance of a frustum.
+      theta_width: A float scalar sets the theta width of a frustum.
+      phi_width: A float scalar sets the phi width of a frustum.
+      max_noise_level: A float scalar sets the probability threshold for dropping the points.
+
+    """
+
+    def __init__(
+        self, r_distance, theta_width, phi_width, max_noise_level=None, **kwargs
+    ):
+        super().__init__(**kwargs)
+
+        if r_distance < 0:
+            raise ValueError("r_distance must be >=0.")
+        if theta_width < 0:
+            raise ValueError("theta_width must be >=0.")
+        if phi_width < 0:
+            raise ValueError("phi_width must be >=0.")
+        max_noise_level = max_noise_level if max_noise_level else 0.0
+        if max_noise_level < 0 or max_noise_level > 1:
+            raise ValueError("max_noise_level must be >=0 and <=1.")
+
+        self._r_distance = r_distance
+        self._theta_width = theta_width
+        self._phi_width = phi_width
+        self._max_noise_level = max_noise_level
+
+    def get_random_transformation(self, point_clouds, **kwargs):
+        # Randomly select a point from the first frame as the center of the frustum.
+        num_valid_points = tf.cast(
+            tf.math.reduce_sum(point_clouds[0, :, POINTCLOUD_LABEL_INDEX]), tf.int32
+        )
+        randomly_select_point_index = tf.random.uniform(
+            (), minval=0, maxval=num_valid_points, dtype=tf.int32
+        )
+        randomly_select_frustum_center = point_clouds[
+            0, randomly_select_point_index, :POINTCLOUD_LABEL_INDEX
+        ]
+        num_frames, num_points, num_features = point_clouds.get_shape().as_list()
+        frustum_mask = []
+        for f in range(num_frames):
+            frustum_mask.append(
+                point_cloud.within_a_frustum(
+                    point_clouds[f],
+                    randomly_select_frustum_center,
+                    self._r_distance,
+                    self._theta_width,
+                    self._phi_width,
+                )[tf.newaxis, :, tf.newaxis]
+            )
+        frustum_mask = tf.concat(frustum_mask, axis=0)
+        feature_noise = 1 + tf.random.uniform(
```

Comment:
What is the purpose of adding `1` to the noise? Do we expect to move all points in the positive direction in each axis?